static/app/components/nav/sidebar.tsx

```diff
@@ -127,7 +135,14 @@ const NavButton = styled('button')`
   ${linkStyles}
 `;
 
-function SidebarLink({children, item}: SidebarItemProps & {children: React.ReactNode}) {
+function SidebarLink({
+  children,
+  item,
+  onClick,
+}: SidebarItemProps & {
+  children: React.ReactNode;
+  onClick: React.HTMLAttributes<HTMLAnchorElement>['onClick'];
+}) {
```

Comment:
The type intersection here is a big no-no :D

I recommend pretty much everyone to read the typescript [performance docs](https://github.com/microsoft/Typescript/wiki/Performance#preferring-interfaces-over-intersections) to understand exactly why.

In summary, TS is unfortunately not very good at intersecting types - the larger the type, the more the compiles struggles (afaik there is also no type caching here, so the more instances you have, the slower things get). These HTMLAttributes types, while seemingly harmless, are actually absolute massive, as they contain all enumerable html attributes, including the entire CSS definition.

A bit of historical context, but about 3 years back when we joined Sentry, things were so bad that making a simple type change took the lsp more than a second to process... You would make a change and it literally took one second or more for the changes to reflect in your IDE. 

We basically halved the time and iirc memory usage of the type checker, mostly by removing intersections like these. There is a [PR](https://github.com/getsentry/sentry/pull/30847) where I did the initial exploration, and then if you search for `ref(tsc)`, there is a bunch of PRs like [this one](https://github.com/getsentry/sentry/pull/32153) where you can see the impact of these changes.

Reply:
TIL—great context to keep in mind, thanks @JonasBa! Fixed in a81116cc1d4